### PR TITLE
HIPP-977: oas discovery proxy

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,67 @@
...
+
+## Dependencies
+Beyond the typical HMRC Digital platform dependencies this service relies on:
+- OAS Discovery API Service
+
+The full set of dependencies can be started using Service Manager and the group API_HUB_ALL.
+
+You can view service dependencies using the Tax Catalogue's Service Relationships
+section here:
+https://catalogue.tax.service.gov.uk/service/identity-management-service-proxy
```
Comment (edgarjimenez) [inline README.md]: should be `https://catalogue.tax.service.gov.uk/service/oas-discovery-proxy`

File: README.md [hunk 2]
```diff
@@ -1,8 +1,67 @@
...
+
+### OAS Discovery API
+All requests to this service starting with are forwarded to IDMS.
```
Comment (edgarjimenez) [inline README.md]: starting with `/oas-discovery-proxy/internal` are forwarded to the `OAS Discovery API`

File: conf/app.routes [hunk 3]
```diff
@@ -1,3 +1,2 @@
 # microservice specific routes
-
-GET        /hello-world             uk.gov.hmrc.oasdiscoveryproxy.controllers.MicroserviceHelloWorldController.hello
+-> / uk.gov.hmrc.oasdiscoveryproxy.controllers.OasDiscoveryRouter
```
Comment (edgarjimenez) [inline conf/app.routes]: should this not be `/internal`, based on the [OAS file](https://github.com/hmrc/integration-catalogue-oas-files/blob/main/platforms/hip/publish-files/oas-discovery-api.yaml#L19)?

File: test/uk/gov/hmrc/oasdiscoveryproxy/controllers/OasDiscoveryControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  with HttpClientV2Support
+  with OptionValues
+  with MockitoSugar {
+
+  "GET request" - {
+    "must be forwarded with headers" in {
+      val responseBody = """{"jam": "scones"}"""
+
+      stubFor(
+        get(urlEqualTo("/oas-discovery-stubs/v1/oas-deployments"))
```
Comment (edgarjimenez) [inline test/uk/gov/hmrc/oasdiscoveryproxy/controllers/OasDiscoveryControllerSpec.scala:48]: As above, this should have _internal_ in the path: `/oas-discovery-stubs/internal/v1/oas-deployments`

File: README.md (hunk 1 above)
Comment (llmikeyj) [reply README.md]: Ok

File: README.md (hunk 2 above)
Comment (llmikeyj) [reply README.md]: Ok

File: conf/app.routes (hunk 3 above)
Comment (llmikeyj) [reply conf/app.routes]: Setting to /v1/oas-deployments as discussed

File: test/uk/gov/hmrc/oasdiscoveryproxy/controllers/OasDiscoveryControllerSpec.scala (hunk 4 above)
Comment (llmikeyj) [reply test/uk/gov/hmrc/oasdiscoveryproxy/controllers/OasDiscoveryControllerSpec.scala:48]: No change required here, as discussed

**Review by edgarjimenez — APPROVED** [review]

**Pull request merged**